fuzzers/libafl-fuzz/src/fuzzer.rs

```diff
@@ -0,0 +1,395 @@
+     * */
+    let mut objective = feedback_or!(
+        feedback_and!(
+            feedback_or_fast!(
+                CrashFeedback::new(),
+                feedback_and!(
+                    ConstFeedback::new(!opt.ignore_timeouts),
+                    TimeoutFeedback::new()
+                )
+            ),
+            MaxMapFeedback::with_name("edges_objective", &edges_observer)
+        ),
+        CustomFilepathToTestcaseFeedback::new(set_solution_filepath, fuzzer_dir.clone())
+    );
+
+    // Initialize our State if necessary
+    let mut state = state.unwrap_or_else(|| {
+        StdState::new(
+            StdRand::with_seed(current_nanos()),
+            // TODO: configure testcache size
+            CachedOnDiskCorpus::<BytesInput>::new(fuzzer_dir.join("queue"), 10).unwrap(),
+            OnDiskCorpus::<BytesInput>::new(fuzzer_dir.clone()).unwrap(),
+            &mut feedback,
+            &mut objective,
+        )
+        .unwrap()
+    });
+
+    // Create our Mutational Stage.
+    let power = StdPowerMutationalStage::new(StdScheduledMutator::new(
+        havoc_mutations().merge(tokens_mutations()),
+    ));
+    let strategy = opt.power_schedule.unwrap_or(PowerSchedule::EXPLORE);
+
+    // Create our ColorizationStage
+    let colorization = ColorizationStage::new(&edges_observer);
+
+    // Create our Scheduler
+    let mut weighted_scheduler =
+        StdWeightedScheduler::with_schedule(&mut state, &edges_observer, Some(strategy));
+    if opt.cycle_schedules {
+        weighted_scheduler = weighted_scheduler.cycling_scheduler();
+    }
+    let scheduler = IndexesLenTimeMinimizerScheduler::new(&edges_observer, weighted_scheduler);
+
+    // Create our Fuzzer
+    let mut fuzzer = StdFuzzer::new(scheduler, feedback, objective);
+
+    // Create the base Executor
+    let mut executor = base_executor(opt, &mut shmem_provider);
+    // Set a custom exit code to be interpreted as a Crash if configured.
+    if let Some(crash_exitcode) = opt.crash_exitcode {
+        executor = executor.crash_exitcode(crash_exitcode);
+    }
+
+    // Enable autodict if configured
+    let mut tokens = Tokens::new();
+    if !opt.no_autodict {
+        executor = executor.autotokens(&mut tokens);
+    };
+
+    // Set a custom directory for the current Input if configured;
+    // May be used to provide a ram-disk etc..
+    if let Some(cur_input_dir) = &opt.cur_input_dir {
+        if opt.harness_input_type.is_none() {
+            return Err(Error::illegal_argument(
+                "cannot use AFL_TMPDIR with stdin input type.",
+            ));
+        }
+        executor = executor.arg_input_file(cur_input_dir.join(get_unique_std_input_file()));
+    }
+
+    // Finalize and build our Executor
+    let mut executor = executor
+        .build(tuple_list!(time_observer, edges_observer))
+        .unwrap();
+
+    // Load our seeds.
+    if state.must_load_initial_inputs() {
+        state
+            .load_initial_inputs(
```

Comment:
we probably want to use the multicore version here, right?

Reply:
Yes